lib/subscriptions_test_kit/suites/subscriptions_r5_backport_r4_server/subscription_rejection/reject_subscription_cross_version_extension_test.rb

```diff
@@ -0,0 +1,71 @@
+      title 'Server Handles Unsupported Cross-Version Extensions'
+      description %(
+        When processing a request for a Subscription a server SHOULD verify that the Subscription is supported and does
+        not contain any information not implemented by the server. If the Subscription is no supported, the server
+        should reject the Subscription create request, or it should attempt to adjust the Subscription. Since the FHIR
+        R5 is currently under development, there are no guarantees these extensions will meet the requirements of
+        this guide. In order to promote widespread compatibility, cross version extensions SHOULD NOT be used
+        on R4 subscriptions to describe any elements.
+
+        The test will pass if the server either
+        1. rejects the Subscription by responding with a non-201 response, or
+        2. updates the Subscription resource to remove or replace the unsupported value.
+      )
+
+      verifies_requirements 'hl7.fhir.uv.subscriptions_1.1.0@12'
+
+      input :subscription_resource,
+            title: 'Workflow Subscription Resource',
+            type: 'textarea',
+            description: %(
+              A Subscription resource in JSON format that Inferno will send to the server under test
+              so that it can demonstrate its ability to perform the Subscription creation and Notification
+              response workflow. The instance must be conformant to the R4/B Topic-Based Subscription profile.
+              Inferno may modify the Subscription before submission, e.g., to point to Inferno's notification endpoint.
+              This input is also used by the unsupported Subscription test as the base on which to add unsupported
+              element values to test for server rejection.
+            )
+
+      run do
+        assert_valid_json(subscription_resource)
+        subscription = JSON.parse(subscription_resource)
+
+        unsupported_info = {
+          'unsupported_title' => 'cross-version extensions',
+          'field_path' => ['_criteria'],
+          'field_value' => { 'extension' => [{
+            url: 'http://hl7.org/fhir/5.0/subscriptions-backport/StructureDefinition/backport-filter-criteria',
+            valueString: 'Encounter?patient=Patient/123'
+          }] }
+        }
+
+        skip_if(unsupported_info['field_value'].blank?, %(
+          No subscription topic input provided.))
+
+        field_name = unsupported_info['field_path'].last
+
+        outer_field_name = unsupported_info['field_path'].first
+        subscription_field = if unsupported_info['field_path'].length > 1
+                               subscription[outer_field_name]
+                             else
+                               subscription
+                             end
+
+        subscription_field[field_name] = unsupported_info['field_value']
+
+        send_unsupported_subscription(subscription, unsupported_info['unsupported_title'],
```

Comment:
This logic does not appear to be doing the right thing. When I run it against the client tests, which echo the request back with a 201, it succeeds.  Seems like that should be a failure because the cross version extension is still in the returned subscription and it succeeded.

Reply:
Addressed in [06d86cb](https://github.com/inferno-framework/subscriptions-test-kit/pull/7/commits/06d86cb11ecdc85f1d8d5b32c62be7cf682edc91). Made changes to this test to actually just fail when receiving a `201` since I don't think it makes sense to alter the values based on the requirement description ("cross version extensions SHOULD NOT be used on R4 subscriptions to describe any elements") from [HL7 guide](https://hl7.org/fhir/uv/subscriptions-backport/STU1.1/components.html#subscriptions). I can change this though to be closer to what it was before, or remove the test if we don't think it's as fleshed out as it doesn't have a requirement.

Reply:
Not sure I agree with the changes. I think both approaches - allowing removal of the offending value are requiring rejection - are defensible. The [IG also says that](https://hl7.org/fhir/uv/subscriptions-backport/STU1.1/components.html#subscriptions-and-fhir-versions) "servers MAY reject or coerce, according to their policies". Its in a different section, but it feels like it could easily apply here as well and that "coerce" could include removal of invalid extensions.  If the verification code was working, I would prefer to leave it in rather than rip it out at this point. Interested to hear your reactions.

Reply:
nevermind - I guess the point was that this logic wasn't working, so I'm fine leaving it out!